batch_convert.qml

```diff
@@ -648,32 +703,43 @@ MuseScore {
       }
 
       var curFileInfo = fileList.shift()
-      var shortName = curFileInfo[0]
+      var filePath = curFileInfo[0]
       var fileName = curFileInfo[1]
-      var fileBase = curFileInfo[2]
+      var fileExt = curFileInfo[2]
+      
+      var fileFullPath = filePath + fileName + "." + fileExt
 
       // read file
-      var thisScore = readScore(fileName,true)
+      var thisScore = readScore(fileFullPath, true)
 
       // make sure we have a valid score
       if (thisScore) {
         // get modification time of source file
-        file.source = fileName
-        var srcModifiedTime = file.modifiedTime()
+        fileScore.source = fileFullPath
+        var srcModifiedTime = fileScore.modifiedTime()
         // write for all target formats
         for (var j = 0; j < outFormats.extensions.length; j++) {
-          var targetFile = fileBase + "." + outFormats.extensions[j]
+          if(differentExportPath.checked) {
+            fileScore.source = targetFolderDialog.folderPath + "/" + fileName + "." + outFormats.extensions[j]
+            }
+          else {
+            fileScore.source = filePath + fileName + "." + outFormats.extensions[j]
+            }
 
           // get modification time of destination file (if it exists)
           // modifiedTime() will return 0 for non-existing files
-          file.source = targetFile
-
           // if src is newer than existing write this file
-          if (srcModifiedTime > file.modifiedTime()) {
-             var res = writeScore(thisScore, targetFile, outFormats.extensions[j])
-             resultText.append("%1 → %2".arg(fileName).arg(outFormats.extensions[j]))
+          if (srcModifiedTime > fileScore.modifiedTime()) {
+             var res = writeScore(thisScore, fileScore.source, outFormats.extensions[j])
+             
+             if(res) {
+               resultText.append("%1 → %2".arg(fileFullPath).arg(outFormats.extensions[j]))
+               }
+             else {
+               resultText.append("Error: %1 → %2 not exported".arg(fileFullPath).arg(outFormats.extensions[j]))
+               }
           } else {
-             resultText.append(qsTr("%1.%2 is up to date").arg(fileBase).arg(outFormats.extensions[j]))
+             resultText.append(qsTr("%1 is up to date").arg(fileScore.source).arg(outFormats.extensions[j]))
```

Comment:
2 `.arg()` need 2 `%n`

Reply:
Haha, runtime error. Sorry for that stupid one.